source/opt/set_spec_constant_default_value_pass.h

```diff
@@ -43,6 +43,46 @@ class SetSpecConstantDefaultValuePass : public Pass {
   const char* name() const override { return "set-spec-const-default-value"; }
   bool Process(ir::Module*) override;
 
+  // Parses the given null-terminated c string to get a mapping from Spec Id to
+  // default value strings. Returns a unique pointer of the mapping from spec
+  // ids to spec constant default value strings built from the given |str| on
+  // success. Returns a nullptr if the given string is not valid for building
+  // the mapping.
+  // A valid string for building the mapping should follow the rule below:
+  //
+  //  { <spec id A> : <default value for A>, <spec id B> : <default value for
```

Comment:
This is a more complex syntax than you needed.  
Since spaces are not allowed inside a number, you could have used space as a delimiter between key/value pairs.
Then you just split on whitespace, and then for each term split on colon.